From a1573330b57756a022e8b4e7f272aa5bf15b852f Mon Sep 17 00:00:00 2001
From: He Zhe <zhe.he@windriver.com>
Date: Tue, 13 Oct 2015 02:26:57 -0400
Subject: [PATCH 2/2] simics: Avoid to access unsupported ARMv8 register
 CPUID_EXT_ISAR5

cpuid_init_hwcaps accesses ARMv8 register CPUID_EXT_ISAR5 which is not
simulated by simics, which causes system hang.
Add CONFIG_SIMICS_A9 to tell if current arch is simics arma9.

Signed-off-by: He Zhe <zhe.he@windriver.com>
---
 arch/arm/kernel/setup.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/kernel/setup.c b/arch/arm/kernel/setup.c
index 6c777e9..6e16dcd 100644
--- a/arch/arm/kernel/setup.c
+++ b/arch/arm/kernel/setup.c
@@ -389,6 +389,7 @@ static void __init cpuid_init_hwcaps(void)
 	if (block >= 5)
 		elf_hwcap |= HWCAP_LPAE;
 
+#if !defined(CONFIG_SIMICS_A9)
 	/* check for supported v8 Crypto instructions */
 	isar5 = read_cpuid_ext(CPUID_EXT_ISAR5);
 
@@ -409,6 +410,7 @@ static void __init cpuid_init_hwcaps(void)
 	block = cpuid_feature_extract_field(isar5, 16);
 	if (block >= 1)
 		elf_hwcap2 |= HWCAP2_CRC32;
+#endif
 }
 
 static void __init elf_hwcap_fixup(void)
-- 
2.0.2

